src/mme-app/msgHandlers/gtpMsgHandler.cpp

```diff
@@ -170,10 +172,11 @@ void GtpMsgHandler::handleDdnMsg_v(IpcEventMessage* eMsg, uint32_t ueIdx)
 {
 	log_msg(LOG_INFO,"Inside handle DDN\n");
 
-	SM::ControlBlock* controlBlk_p = SubsDataGroupManager::Instance()->findControlBlock(ueIdx);
+	SM::ControlBlock* controlBlk_p =
+	        MmeCommonUtils::findControlBlockForS11Msg(eMsg->getMsgBuffer());
 	if(controlBlk_p == NULL)
 	{
-		log_msg(LOG_ERROR, "handleReleaseBearerResponse_v: "
+		log_msg(LOG_ERROR, "handleDdnMsg_v: "
```

Comment:
add enhancement defect to reject DDN with cause UNKNOWN IMSI. This helps SGW to cleanup the session.

Reply:
Here Control block will be NULL only if ue index in DDN message is 0. All other cases will be handled in default ddn handler, where we will respond with 'ue context not found' cause.